src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -226,57 +227,92 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
+     getWorkspaceContext(workspaceName) flatMap { ctx =>
+      deleteWorkspace(workspaceName, ctx)
+    }
+  }
 
-  private def deleteWorkspace(workspaceName: WorkspaceName, dataAccess: DataAccess, workspaceContext: SlickWorkspaceContext): ReadWriteAction[PerRequestMessage] = {
+  private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: SlickWorkspaceContext): Future[PerRequestMessage] = {
+    def deleteGroups(dataAccess: DataAccess): Iterable[ReadWriteAction[Boolean]] = {
+    workspaceContext.workspace.accessLevels.map { case (_, group) =>
+      dataAccess.rawlsGroupQuery.delete(group)}
+    }
     //Attempt to abort any running workflows so they don't write any more to the bucket.
     //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
     //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
     //ExecutionContext run the futures whenever
-    dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { recs => recs.map {
-      case wf if !WorkflowStatuses.withName(wf.status).isDone && wf.externalId.isDefined => executionServiceCluster.abort(wf, userInfo).map {
-        case Failure(regrets) =>
-          logger.info(s"failure aborting workflow ${wf.externalId} while deleting workspace ${workspaceName}", regrets)
-          Failure(regrets)
-        case success => success
-      }
-    }} andThen {
-      DBIO.from(gcsDAO.deleteBucket(workspaceContext.workspace.bucketName, bucketDeletionMonitor))
-    } andThen {
-      val groupRefs: Set[RawlsGroupRef] = workspaceContext.workspace.accessLevels.values.toSet ++ workspaceContext.workspace.realmACLs.values
-      DBIO.seq(groupRefs.map { groupRef => dataAccess.rawlsGroupQuery.load(groupRef) flatMap {
-        case Some(group) => DBIO.from(gcsDAO.deleteGoogleGroup(group))
-        case None => DBIO.successful(Unit)
-      }}.toSeq: _*)
-    } andThen {
-      DBIO.seq(dataAccess.workspaceQuery.deleteWorkspaceAccessReferences(workspaceContext.workspaceId))
-    } andThen {
-      DBIO.seq(dataAccess.workspaceQuery.deleteWorkspaceSubmissions(workspaceContext.workspaceId))
-    } andThen {
-      DBIO.seq(dataAccess.workspaceQuery.deleteWorkspaceMethodConfigs(workspaceContext.workspaceId))
-    } andThen {
-      dataAccess.workspaceQuery.deleteWorkspaceEntitiesAndAttributes(workspaceContext.workspaceId)
-    } andThen {
-      DBIO.seq(workspaceContext.workspace.accessLevels.map { case (_, group) =>
-        dataAccess.rawlsGroupQuery.delete(group)
-      }.toSeq: _*)
-    } andThen {
-      dataAccess.workspaceQuery.delete(workspaceName)
-    } andThen {
-      DBIO.successful(RequestComplete(StatusCodes.Accepted, s"Your Google bucket ${workspaceContext.workspace.bucketName} will be deleted within 24h."))
+    val deletionFuture: Future[(Seq[WorkflowRecord], String, Seq[Option[RawlsGroup]])] = dataSource.inTransaction { dataAccess => {
```

Comment:
I don't think you need the extra brace after `dataAccess =>` here? (Which means you don't need the }} on 296 either, which is what made me notice this in the first place)